.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+# name: CI
+
+# on:
+#   push:
+#     branches:
+#       - $default-branch
+#     tags:
+#       - v*
+#   pull_request:
+
+# # TODO: do we need this?
+# # permissions:
+# #   contents: read # for actions/checkout to fetch code
+
+# # TODO: verify-codegen
+
+# jobs:
+#   test:
+#     runs-on: ubuntu-latest
+#     strategy:
+#       fail-fast: false
+#       matrix:
+#         go-version:
+#           - "1.19"
+#           - "1.20"
+#     name: "${{ matrix.go-version }}"
+#     steps:
+#       - uses: actions/checkout@v3
+
+#       - name: Setup Go ${{ matrix.go-version }}
+#         uses: actions/setup-go@v4
+#         with:
+#           go-version: ${{ matrix.go-version }}
+
+#       - name: Build
+#         run: make manager
+
+#       - name: Test
+#         run: make test
+
+#   golangci:
+#     permissions:
+#       pull-requests: read # for golangci/golangci-lint-action to fetch pull requests
+#     runs-on: ubuntu-latest
+#     steps:
+#       - uses: actions/checkout@v3
+
+#       - uses: actions/setup-go@v4
+#         with:
+#           go-version: "1.20"
+
+#       - uses: golangci/golangci-lint-action@v3
+#         with:
+#           version: latest
+#           only-new-issues: true
+
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Build
+        run: go build -v ./...
```

Comment:
Is building necessary here? Wouldn't `go test` already reveal possible compile errors?